src/posthog-core.ts

```diff
@@ -1074,6 +1105,8 @@ export class PostHog {
             this.register({ distinct_id: new_distinct_id })
         }
 
+        // historically we checked for a missing device id *or* distinct_id == device_id as a proxy for detecting anonymous users
+        // we now have the explicit anonymous user state but we still check for a missing device id to maintain backwards compatibility
```

Comment:
dumb q: why do we need backwards compatibility here? 

If they update the library they just get the new state tracking stuff, wouldn't they. If they use  an older version of the library the code had the missing deviceID code